### PR TITLE
EspressJS to Fastify migration

### PR DESCRIPTION
**Pull request by giulianok**
Express to Fastify migration that will improve overall performance on requests/responses including SSR 

## Description
The goal with this PR is to run all One App servers under Fastify while supporting existing ExpressJS code to slowly migrate to Fastify. This is done through a library called `@fastify/express` that allows us run ExpressJS through Fastify.

## Motivation and Context

Fastify handles more requests per second than ExpressJS. It is also better maintained, provides a cleaner syntax, encapsulation, and more. To minimize risk, in this PR, we are introducing Fastify + Express as described in the **Description** section above.

### Benchmark from Fastify website
Screen Shot 2022-07-26 at 1 18 22 PM

### Benchmark local with fastify-express
Screen Shot 2022-06-21 at 1 12 11 PM (1)

The metrics still show that fastify handles 4x the amount of requests compared to express, and fastify-express is slower than fastify but still 3x faster than express

## How Has This Been Tested?

Tests were performed via manual tests, unit testing, and integration tests with v5 and v6 versions.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

There's no impact in DX at the moment

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -23,109 +24,134 @@ import './init';
...
+    ? Object.values(require(pathToDevEndpoints)()).reduce(
+      (moduleRemotes, { destination, devProxyPath }) => ({
+        ...moduleRemotes,
+        [devProxyPath]: destination,
+      }),
+      {})
+    : {};
+};
+
+export const listen = async ({
```
Comment (JAdshead) [inline src/server/index.js:48]: are you planning on moving back to `./listen`

**Review by JAdshead — CHANGES_REQUESTED** [review]

File: src/server/index.js (hunk 1 above)
Comment (giulianok) [reply src/server/index.js:48]: I am not since this is a fn only consumed in `index.js`

File: __tests__/server/ssrServer.spec.js [hunk 2]
```diff
@@ -55,7 +57,7 @@ jest.mock('../../mocks/scenarios', () => ({
 }), { virtual: true });
 
 describe('ssrServer', () => {
-  jest.spyOn(console, 'info').mockImplementation(() => {});
+  // jest.spyOn(console, 'info').mockImplementation(() => {});
```
Comment (10xLaCroixDrinker) [inline __tests__/server/ssrServer.spec.js:60]:
```suggestion
```

File: __tests__/server/utils/getHttpsConfig.spec.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 American Express Travel Related Services Company, Inc.
```
Comment (10xLaCroixDrinker) [inline __tests__/server/utils/getHttpsConfig.spec.js:2]:
```suggestion
 * Copyright 2022 American Express Travel Related Services Company, Inc.
```

File: src/server/ssrServer.js [hunk 4]
```diff
@@ -121,12 +123,25 @@ export function createApp({ enablePostToModuleRoutes = false } = {}) {
...
+
+  await fastify.register(fastifyExpress);
+
+  fastify.express.disable('x-powered-by');
+  fastify.express.disable('e-tag');
+
+  fastify.use(makeExpressRouter());
 
-  app.use(serverError);
+  // TODO: Needs refactoring (priority)
```
Comment (10xLaCroixDrinker) [inline src/server/ssrServer.js:141]: Is this TODO for this PR?

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

File: src/server/ssrServer.js (hunk 4 above)
Comment (giulianok) [reply src/server/ssrServer.js:141]: no, this is for when we migrate the express routes to fastify. Server errors are being handled from the express router but we still need a 404 handler on the top level

File: __tests__/server/ssrServer.spec.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jest/no-disabled-tests */
```
Comment (JAdshead) [inline __tests__/server/ssrServer.spec.js:1]:
```suggestion
```

File: __tests__/server/ssrServer.spec.js [hunk 6]
```diff
@@ -393,80 +437,58 @@ describe('ssrServer', () => {
...
     });
 
     it('should add frame options header for all get calls', async () => {
-      await request(loadServer())
+      await request(loadExpress())
         .get('/route');
       expect(addFrameOptionsHeader).toBeCalled();
     });
 
-    it('should return the static error page when an error was encountered', async () => {
```
Comment (JAdshead) [inline __tests__/server/ssrServer.spec.js:446]: is this still being tested elsewhere ?

Comment (giulianok) [reply __tests__/server/ssrServer.spec.js:446]: it's covered by fastify but I added it back

**Review by JAdshead — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**